yjit/src/yjit.rs

```diff
@@ -96,6 +96,13 @@ fn rb_bug_panic_hook() {
 /// NOTE: this should be wrapped in RB_VM_LOCK_ENTER(), rb_vm_barrier() on the C side
 #[no_mangle]
 pub extern "C" fn rb_yjit_iseq_gen_entry_point(iseq: IseqPtr, ec: EcPtr) -> *const u8 {
+    // Reject very long ISEQs so that we can use u16 instruction indices internally.
+    // Very long ISEQs are also likely to be initialization code.
+    let iseq_size = unsafe { get_iseq_encoded_size(iseq) };
+    if iseq_size > u16::MAX as u32 {
+        return std::ptr::null();
```

Comment:
It might be useful to generate a side exit and count it when `--yjit-stats` is given.

Reply:
Welp, the numbers I got are:

Railsbench
u32: 8,320,851
u16: 8,313,913

It seems like it should be a space saving, but maybe not worth the trouble.